<launch>

  <!-- update source location for collecting data -->
  <arg name="collect_data" default="false" />
  <node name="interactive_adjust_shelf"
        pkg="jsk_interactive_marker" type="marker_6dof">
    <rosparam command="load" file="$(find jsk_arc2017_baxter)/config/collect_data/shelf_marker.yaml"/>
  </node>
  <node name="publish_bin_boxes"
        pkg="jsk_recognition_utils" type="bounding_box_array_publisher.py">
    <rosparam command="load" file="$(find jsk_arc2017_common)/config/shelf_bin.yaml" />
  </node>

  <group ns="fuse_depth_images">
    <node name="output_view"
          pkg="image_view" type="image_view">
      <remap from="image" to="output" />
      <rosparam>
        gui: false
        colormap: 2  # JET
        min_image_value: 0.5
        max_image_value: 1.0
      </rosparam>
    </node>
  </group>

  <!-- *********************************************************************************** -->
  <!-- Data collection                                                                     -->
  <!-- *********************************************************************************** -->

  <group ns="right_hand">
    <node name="data_collection_server"
          pkg="jsk_data" type="data_collection_server.py"
          clear_params="true" output="screen">
      <rosparam subst_value="true">
        save_dir: ~/.ros/jsk_arc2017_baxter/dataset_jsk_v3
        topics:
          - name: /right_hand_left_camera/rgb/camera_info
            msg_class: sensor_msgs/CameraInfo
            fname: camera_info_right_hand_left_camera.yaml
            savetype: YAML
          - name: /fuse_rgb_images/output
            msg_class: sensor_msgs/Image
            fname: image.jpg
            savetype: ColorImage
          - name: /fuse_depth_images/output
            msg_class: sensor_msgs/Image
            fname: depth.npz
            savetype: DepthImage
          - name: /fuse_depth_images/output_view/output
            msg_class: sensor_msgs/Image
            fname: depth_viz.jpg
            savetype: ColorImage
          - name: /right_hand/tf_to_transform/output
            msg_class: geometry_msgs/TransformStamped
            fname: tf_camera_rgb_from_base.yaml
            savetype: YAML
        params:
          - key: /right_hand/target_bin_name
            fname: shelf_bin.txt
            savetype: Text
          - key: /right_hand/view_frame
            fname: view_frame.txt
            savetype: Text
      </rosparam>
    </node>

    <node name="tf_to_transform"
          pkg="jsk_topic_tools" type="tf_to_transform.py">
      <rosparam>
        parent_frame_id: base
        child_frame_id: right_hand_left_camera_rgb_optical_frame
        duration: 1
        rate: 30
      </rosparam>
    </node>
  </group>

  <node name="collect_data_in_shelf_main"
        pkg="jsk_arc2017_baxter" type="collect-data-in-shelf-main.l"
        output="screen">
  </node>

  <node name="rqt_yn_btn"
        pkg="jsk_rqt_plugins" type="rqt_yn_btn">
    <remap from="~" to="/rviz/yes_no_button" />
  </node>

</launch>
